todo_app/app.py

```diff
@@ -9,39 +9,63 @@
 import requests
 import os
 
+def create_app():
+    app = Flask(__name__)
+    app.config.from_object(Config())
 
-app = Flask(__name__)
-app.config.from_object(Config())
-
-
-@app.route('/')
-def index():
-
-    return render_template('index.html', result = trello_get_items())
-
-@app.route('/add', methods=['POST'])
-def add():
-    if not request.form['title'].isspace():
-        add_result = trello_add_item(request.form['title'], request.form['desc'], request.form['datepicker'])
-        return render_template('index.html', result_message=add_result, result = trello_get_items())
-    else:
-        return render_template('index.html', result_message="No title given", result = trello_get_items())
-
-@app.route('/complete', methods=['GET'])
-def complete():
-    error = None
-    if not request.args.get('id','').isspace():
-        complete_result = trello_complete_item(request.args.get('id',''))
-        return render_template('index.html', result_message=complete_result, result = trello_get_items())
-    else:
-        return render_template('index.html', result_message="didnt get an item id", result = get_items())
-
-@app.route('/reset', methods=['GET'])
-def reset():
-    error = None
-    if not request.args.get('id','').isspace():
-        reset_result = trello_todo_item(request.args.get('id',''))
-        return render_template('index.html', result_message=reset_result, result = trello_get_items())
-    else:
-        return render_template('index.html', result_message="didnt get an item id", result = get_items())
 
+    @app.route('/')
+    def index():
+        item_view_model = ViewModel(items = trello_get_items(), result_message=None)
+        return render_template('index.html', view_model=item_view_model)
+
+    @app.route('/add', methods=['POST'])
+    def add():
+        if not request.form['title'].isspace():
+            result_message = trello_add_item(request.form['title'], request.form['desc'], request.form['datepicker'])
+        else:
+            result_message = "No title given"
+        item_view_model = ViewModel(trello_get_items(), result_message)
+        return render_template('index.html', view_model=item_view_model)
+
+    @app.route('/complete', methods=['GET'])
+    def complete():
+        if not request.args.get('id','').isspace():
```

Comment:
This doesn't do what you think it does.

`''.isspace()` is False.

So if id is not set at all, this if condition is True.